components/script/dom/element.rs

```diff
@@ -2203,6 +2203,19 @@ impl VirtualMethods for Element {
             return;
         }
 
+        if self.is_focusable_area() {
+            let doc = document_from_node(self);
+            let children = self.upcast::<Node>().traverse_preorder();
+
+            for child in children {
+                if let Some(element) = child.downcast::<Element>() {
+                    if element.is_focusable_area() && !element.disabled_state() {
+                        doc.add_focusable_element(element);
+                    }
+                }
+            }
+        }
```

Comment:
When do you remove focusable elements from the document? Shouldn't that be done in `unbind_from_tree`?